src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -87,6 +78,29 @@ public boolean executeAndWait(Callable command) {
         }
     }
 
+    /**
+     * This method allows to execute a callable task that provides a return value after the calculation is done.
+     * 
+     * @param command The task to execute.
+     * @return A Future object that provides the returning value.
+     */
+    public <T> Future<T> executeAndReturn(Callable<T> command) {
+        Objects.requireNonNull(command);
+        return executorService.submit(command);
+    }
+    
+    public <T> List<Future<T>> executeAll(Collection<Callable<T>> tasks) {
+        Objects.requireNonNull(tasks);
+        List<Future<T>> futures = null; 
+        try {
+            futures =  executorService.invokeAll(tasks);
+        } catch (InterruptedException e) {
+            LOGGER.info("Invokation has been interrupted during execution.");
+        }
+        Objects.requireNonNull(futures);
```

Comment:
This is asserted by the API. I think, this line can be removed.

Typically `Objects.requireNonNull(...)` is only used at the beginning of a method to check parameters. Not in the middle.

Reply:
I just wanted to avoid returning null. I'll remove it.

Reply:
Yeah, totally understandable. I think, that's why `@NonNull` annotations are there for... --> https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_annotations.htm We are currently not using them in JabRef.

Otherwise, the whole could had to be cluttered with `Objects.requireNonNull(...)` statements, because `null` should be avoided in any case. 🔢